app/models/manageiq/providers/kubevirt/inventory/parser.rb

```diff
@@ -212,8 +214,49 @@ def process_template(object)
     template_object.vendor = ManageIQ::Providers::Kubevirt::Constants::VENDOR
 
     # Add the inventory object for the hardware:
+    process_hardware(template_object, object.spec.template.spec.domain, metadata)
+
+    # Add the inventory object for the OperatingSystem
+    process_os(template_object, metadata)
+  end
+
+  def process_hardware(template_object, domain, metadata)
     hw_object = hw_collection.find_or_build(template_object)
-    hw_object.memory_mb = ManageIQ::Providers::Kubevirt::MemoryCalculator.convert(domain.memory, 'Mi')
+    hw_object.memory_mb = ManageIQ::Providers::Kubevirt::MemoryCalculator.convert(domain.resources.requests.memory, 'Mi')
+    hw_object.cpu_cores_per_socket = domain.cpu.cores
+    hw_object.cpu_total_cores = domain.cpu.cores
+    hw_object.guest_os = metadata.labels.send("miq.github.io/kubevirt-os")
```

Comment:
```metadata.labels.send("miq.github.io/kubevirt-os")``` what is the expected result ? shouldn't send receive a symbol ?

Reply:
KubeClient is using `open_struct`s and we can call methods on yaml structure like we would have an object. Here the trick is that `miq.github.io/kubevirt-os` is not valid method name so we use `send` as workaround.